### PR TITLE
fix(techdocs): Fixes anchor bug in Techdocs reader 

### LINKED ISSUE #15597: 🐛 Bug Report: Techdocs anchor jump not working on page reload
### 📜 Description

In our backstage deployment. The techdocs reader often does not jump to the anchor provided in the URL hash. 

### 👍 Expected behavior

a techodocs page with a hash in the URL should jump to the anchor in the document on load

### 👎 Actual Behavior with Screenshots

a techodocs page with a hash in the URL sometimes jumps to the wrong location on load

### 👟 Reproduction steps

1. In a backstage deployment with slow connection
2. Open a techdocs page with anchors
3. Click on an anchor
4. Reload the page
5. The page jumps to the wrong location

### 📃 Provide the context for the Bug.

_No response_

### 🖥️ Your Environment

_No response_

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by gtestault**
Fixed a bug in techdocs reader where loading a techdocs page with a hash did not always result in a jump to the right anchor location of the document.

fixes #15597

Signed-off-by: Gabriel Testault <gabriel.testault@goto.com>

## Hey, I just made a Pull Request!

- The scrollIntoAnchor transformer now jumps to the anchor only after css was loaded instead of jumping after 200ms. Implemenation similar to onCssLoaded
- added tests for new behavior
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/techdocs/src/reader/transformers/scrollIntoAnchor.ts [hunk 1]
```diff
@@ -15,17 +15,24 @@
...
-        // fix invalid selector error for anchor starting with number
-        dom?.querySelector(`[id="${hash}"]`)?.scrollIntoView();
-      }
-    }, 200);
+    dom.addEventListener(
+      SHADOW_DOM_STYLE_LOAD_EVENT,
+      function handleShadowDomStyleLoad() {
+        if (window.location.hash) {
+          const hash = window.location.hash.slice(1);
+          // fix invalid selector error for anchor starting with number
```
Comment (agentbellnorm) [inline plugins/techdocs/src/reader/transformers/scrollIntoAnchor.ts:27]: I know this comment is old, but I don't really understand it, do you? Like Is it a todo or is it describing the row beneath? Maybe we can delete it or clarify it if possible.

**Review by agentbellnorm — APPROVED** [review]
Cool! This seems like the proper solution 👍. Had one minor comment/question.

File: plugins/techdocs/src/reader/transformers/scrollIntoAnchor.ts (hunk 1 above)
Comment (gtestault) [reply plugins/techdocs/src/reader/transformers/scrollIntoAnchor.ts:27]: we can delete this comment IMO, it refers to implementation details for a fix in the previous commit that changed the queryselector from: ``dom?.querySelector(`#${hash}`)``  to ``dom?.querySelector(`[id="${hash}"]`)``in commit f90ffdc8

**Comment by gtestault** [comment]
so, can we merge this?

**Pull request merged**